### PR TITLE
UI: Add button to fail running deployments

### PR DESCRIPTION
**Pull request by backspace**
As suggested in #9826. As a GIF from [the preview deployment](https://nomad-storybook-and-ui-h2bp8yhvh.vercel.app/ui/jobs/ftp-interface-1):

fail-deployment

This necessitated some customisation options for `TwoStepButton`. One is `inlineText`, which puts the confirmation text in the same line as the buttons. Also, there was a single-use configuration option named `isInfoAction` that I removed in favour of passing a set of class configuration options like this:

```
                @classes={{hash
                  idleButton="is-warning"
                  confirmationMessage="inherit-color"
                  cancelButton="is-danger is-important"
                  confirmButton="is-warning"}}
```

This seemed preferable to me to adding another opaquely-named flag that bundles together a set of class overrides for each part of the button, but I’m open to other approaches!

The new two-step-button stories are [here](https://nomad-storybook-and-ui-h2bp8yhvh.vercel.app/storybook/?path=/story/components-two-step-button--styled) and [here](https://nomad-storybook-and-ui-h2bp8yhvh.vercel.app/storybook/?path=/story/components-two-step-button--inline-text) in the preview deployment.

### CODE REVIEW

**Review by DingoEatingFuzz — CHANGES_REQUESTED** [review]
This is a welcome addition. I have a couple thoughts, dunno how blocking they are, but I think they're at least worth talking through.

1. The `padding-bottom` specific value seems brittle. Ideally we can achieve the baseline matching in a way that still works if we change the button styles or button size.
2. Changing to the classes hash (which is a neat implementation, btw) really opens up the surface area of the component's API. I'm not saying it's wrong, but it gave me pause.

File: ui/app/styles/components/two-step-button.scss [hunk 1]
```diff
@@ -20,6 +20,23 @@
       right: 0;
     }
 
+    &.has-text-inline {
+      top: unset;
+      bottom: 0;
+      padding-bottom: 6.462px;
```
[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/styles/components/two-step-button.scss:26]: yeah, that was admittedly ridiculous 😳 I couldn’t get those to work but what do you think about this?

```
  &.has-inline-text {
    display: inline-flex;
    align-items: center;

    button {
      margin-left: 0.3rem;
    }

    .confirmation-text {
      position: static;
      margin-right: 0.2rem;
    }
  }
```

I’ve learned while looking at this that the spaces between the button are true spaces, not CSS spaces, but with flexbox I added the `0.3rem` margins to bring back the space 🧐

**Comment by backspace** [comment]
agreed that `@classes` is a significant expansion of the interface; my approach was driven by the way `isInfoAction` was handled:

image

I was reluctant to add another set of conditionals in each of those classes, but they could be extracted to computed properties instead. It seemed nice to me to be able to use `TwoStepButton` and style its components independently without having to change anything inside the component itself.

File: ui/app/styles/components/two-step-button.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (DingoEatingFuzz) [reply ui/app/styles/components/two-step-button.scss:26]: I dig it! There is also the hip new `ch` unit that is based on character width that may be worth experimenting with. With a variable width font I don't think this is as simple as `1ch`, you'd probably still need some fraction. And at that point maybe it doesn't even matter... I dunno.

**Review by DingoEatingFuzz — APPROVED** [review]
Reading through the `@classes` approach, I'm more comfortable with it this time. I think your summary of it being styling the individual pieces without having to fuss with structure sold me on it.

My root concern was opening up the API to the point where we aren't really providing a constraint in the way a style guide/design system should. But since the structure and components are fixed and only the individual modifiers are consumer-editable I think this is good 👍

**Comment by backspace** [comment]
Thanks for the rereview! 💞

I `ch`anged to `ch` 😆

I encountered [this test failure](https://app.circleci.com/pipelines/github/hashicorp/nomad/14726/workflows/f4ae0bf2-0699-4d18-b55e-5221aafe393c/jobs/137128) but it’s surely unrelated and it disappeared when I reran, so I’ll look at it separately.

**Pull request merged**